### PR TITLE
Eval like match

### PR DESCRIPTION
**Pull request by Davleen14**
Eval like,match

### CODE REVIEW

File: pkg/ast/spl/spl.peg [hunk 1]
```diff
@@ -650,6 +650,36 @@ BoolExprLevel1 <- "(" first:BoolExprLevel4 ")" {
...
+         LeftValue: stringr.(*structs.ValueExpr),
+         RightValue: pattern.(*structs.ValueExpr),
+         ValueOp: "like",
+     }
+     return expr, nil
+ }
+/ "match(" stringr:ValueExpr ", " pattern:ValueExpr ")" {
+     expr := &structs.BoolExpr {
+         IsTerminal: true,
+         LeftValue: stringr.(*structs.ValueExpr),
```
Comment (AndrewHess) [inline pkg/ast/spl/spl.peg:677]: typo: `stringr` instead of `string`

File: pkg/segment/structs/evaluationstructs.go [hunk 2]
```diff
@@ -272,6 +306,13 @@ func (self *BoolExpr) Evaluate(fieldToValue map[string]utils.CValueEnclosure) (b
 	}
 }
 
+func likePatternToRegex(pattern string) string {
+	pattern = regexp.QuoteMeta(pattern)
+	pattern = strings.Replace(pattern, "%", ".*", -1)
+	pattern = strings.Replace(pattern, "_", ".", -1)
+	return "^" + pattern + "$"
+}
```
Comment (AndrewHess) [inline pkg/segment/structs/evaluationstructs.go:314]: Are you calling this function anywhere?

**Review by AndrewHess — COMMENTED** [review]
Looks pretty good, but you need to add some unit tests

File: pkg/segment/structs/evaluationstructs.go (hunk 2 above)
Comment (Davleen14) [reply pkg/segment/structs/evaluationstructs.go:314]: I have incorporated the functionality directly in the case so no need of this function anymore.

File: pkg/ast/spl/spl.peg (hunk 1 above)
Comment (Davleen14) [reply pkg/ast/spl/spl.peg:677]: Will make the change

File: pkg/segment/structs/evaluationstructs.go (hunk 2 above)
Comment (AndrewHess) [reply pkg/segment/structs/evaluationstructs.go:314]: Ok, just make sure to delete this function then

File: pkg/segment/structs/evaluationstructs_test.go
Comment (AndrewHess) [inline pkg/segment/structs/evaluationstructs_test.go:1]: FYI for these tests you don't need the expressions doing an `if` statement (e.g., the `likeConditionExpr`). You can just make the expression doing the actual check (e.g., the `likeBoolExpr`) and evaluate that to a boolean and assert the value you get back. You can keep it how you have it though.

**Review by AndrewHess — APPROVED** [review]

**Pull request merged**